### ISSUE TITLE
Update Time-dependent Tests to use MockDate

### ISSUE THREAD

**Issue by emilyrohrbough** [cerner/terra-framework#1202]
# Feature Request

## Description

Update WDIO tests that have a date displayed to use a [mockDate](https://www.npmjs.com/package/mockdate) to prevent the screenshots from being affected by the timezone defined in the test environment.

**Comment by supreethmr** [comment]
After investigating the components that have date and time displayed in tests. I think `mockdate` library is not required since most of our test components uses custom date-time values and less dependent on DateTime object or moment object. 

We can prevent change in date format that happens when timezone changes in test environment with following methods : 

- Using moment object `moment.tz.setDefault('America/Chicago');` in tests to set default timeZone.

- Using new `timeZone` prop that is being added to date-time-picker to set timeZone in test pages.

so I would like to know if am missing something which makes `mockdate` better fit for handling timezone changes. 

@emilyrohrbough  @benbcai

**Comment by benbcai** [comment]
@supreethmr I remember I briefly tested with `moment.tz.setDefault('America/Chicago')` a while back and it wasn't working for me the way I expected. I set the timezone on my device to a different timezone (say America/Los_Angeles) and set `moment.tz.setDefault('America/Chicago')`. The time that I got was still in the America/Los_Angeles timezone instead of the Chicago timezone. You can give that a try to see how it works for you.

**Comment by supreethmr** [comment]
I tried setting timeZone to America/Chicago using `moment.tz.setDefault('America/Chicago');` and set the my device timezone to America/New York. Time values were displaying to timezone  `America/Chicago` as shown in below screenshot.

<img width="1680" alt="Screen Shot 2020-09-01 at 7 37 19 AM" src="https://user-images.githubusercontent.com/12869809/91862024-1549f880-ec8b-11ea-85e3-0f28964d86d4.png">

I believe both solution `mockdate.set()` and `moment.tz.setDefault('');` prevents date-time value from getting changed  on timezone update. So I was thinking not to add new library for mocking date. 

I am still not able to find the use-case where mock-date would be really needed by looking into some of the existing implementation of mock-date in consumer projects. Hence thought of using existing code for setting timezone.

**Comment by supreethmr** [comment]
the date set using mock date will remain same on all timezones but UTC part in displayed ISO string changes with timezone even after passing complete date-time value like below to mock date. 
```js
 mockdate.set('2017-08-08T10:30:00+05:30');
```
![date_picker_timeZone1](https://user-images.githubusercontent.com/12869809/92064854-0878f200-edbc-11ea-872b-8d41f99d20c3.gif)

mock date would not be able to control the UTC / GMT value of displayed ISO time value in date-time-picker test pages. 
setting / defaulting the timezone to specific timezone would keep the value unchanged on timezone update. As shown below after setting timeZone to `America/Chicago` ISO displayed value remains same for different timezones.

![date_picker_timeZone2](https://user-images.githubusercontent.com/12869809/92065112-9e148180-edbc-11ea-852c-a33152743c32.gif)

So I am thinking we might need to use the new [timezone prop](https://github.com/cerner/terra-framework/pull/1044) that is being added to date-time-picker to define the timezone for test pages so that date-time remains unchanged for different timeZones.
